src/Http/Uri.php

```diff
@@ -303,6 +303,27 @@ public static function index(array $props = []): static
 		return new static($url, $props);
 	}
 
+	/**
+	 * Inherit query, params and fragment from a parent Uri
+	 * @since 5.2.0
+	 * @return $this
+	 */
+	public function inherit(Uri|string $parent): static
```

Comment:
If someone has a better suggestion for the name, I am not too sure about `inherit`.

Reply:
Some ideas in a rough brainstorming: `integrate`, `succeed`, `combine`, `coalesce`

Reply:
You know what. While reading through the code, I really started to like inherit. Let's keep it.